contracts/bonding/BondingManager.sol

```diff
@@ -566,80 +541,76 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
         }
     }
 
-    /*
-     * @dev Claim token pools shares for a delegator from its lastClaimTokenPoolsSharesRound through the end round
+    /**
+     * @dev Claim token pools shares for a delegator from its lastClaimRound through the end round
      * @param _endRound The last round for which to claim token pools shares for a delegator
      */
-    function claimTokenPoolsShares(uint256 _endRound) external {
+    function claimEarnings(uint256 _endRound) external whenSystemNotPaused currentRoundInitialized {
         // End round must be after the last claim round
-        require(delegators[msg.sender].lastClaimTokenPoolsSharesRound < _endRound);
+        require(delegators[msg.sender].lastClaimRound < _endRound);
         // End round must not be after the current round
         require(_endRound <= roundsManager().currentRound());
 
-        updateDelegatorWithTokenPoolsShares(msg.sender, _endRound);
+        updateDelegatorWithEarnings(msg.sender, _endRound);
     }
 
-    /*
-     * @dev Returns pending bonded stake for a delegator. Includes reward pool shares since lastClaimTokenPoolsSharesRound
+    /**
+     * @dev Returns pending bonded stake for a delegator from its lastClaimRound through an end round
      * @param _delegator Address of delegator
+     * @param _endRound The last round to compute pending stake from
      */
-    function pendingStake(address _delegator) public view returns (uint256) {
+    function pendingStake(address _delegator, uint256 _endRound) public view returns (uint256) {
```

Comment:
`pendingStake` and `pendingFees` can be viewed as the read-only counterparts to `claimEarnings`. While `claimEarnings` will compute the claimable rewards/fees for each round and credit the caller with those values, `pendingStake` and `pendingFees` just compute claimable rewards/fees for each round and what the caller's stake/collected fees would be if it called `claimEarnings`. This `endRound` parameter just allows the caller to configure which round to compute the pending rewards and fees up through